### PR TITLE
WD-1707: Add date selector to API selector

### PR DESCRIPTION
**Pull request by ShaunVanDyk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,30 @@ For example, you may have user 'Admin', who requires access to api:imagery and a
 
 Most of the client implementation is the same between API Key and Client Credentials (i.e. create the leaflet app, register a layer etc.). Even the URL is very similar, instead of passing an api_key parameter, you pass an access_token parameter. And you don't know the API key ahead of time, you need to make a request to your server to get a token. The only other note to make about this is that we get the expiry time of the current token in seconds, so we register a timeout to be called when the token expires to go and grab a new one.
 
+# Product Selector
```
Comment (scottgerring) [inline README.md]: Shouldn't use `product` here. Product references a single tileset, the right word is `composite product`. 
That's probably a bit confusing for 3rd parties though, so perhaps lets use, say, `imagery set` and `date selector`, something like:

`There is a date selector which demonstrates how to get all of the available imagery sets for a particular area. Most of the work is done in the imagery set service on the server side`. 

Which probably leads into naming the product service differently also!

File: README.md [hunk 2]
```diff
@@ -75,6 +75,30 @@ For example, you may have user 'Admin', who requires access to api:imagery and a
...
+
+There is a product selector which demos how to get all the available products for a particular area. Most of the work is done in the product service on the server side.
+
+## Demo bounds
+
+We are limiting the products displayed to be those that cover Perth, and therefore have locked the map to show the region around Perth. There are further considerations when implementing this over a larger area, such as checking if these products still cover the users current view, which is explained in more detail in the clients section.
```
Comment (scottgerring) [inline README.md]: which ~~is~~ **are** explained in more detail

Assuming that you are explaining the set of further considerations in the clients section, and not just how to check if the products still cover the users view?

File: README.md [hunk 3]
```diff
@@ -75,6 +75,30 @@ For example, you may have user 'Admin', who requires access to api:imagery and a
...
+
+## Server implementation
+
+All the work for retrieving products is done in productService.ts. There are a few important steps:
```
Comment (scottgerring) [inline README.md]: all the work for retrieving ~~products~~ **available dates** 

maybe?

File: README.md (hunk 1 above)
Comment (ShaunVanDyk) [reply README.md]: It depends. I do agree, but they should also be using this in the context of the API documents? It talks about composite products there, and shortens things to just products in code. Should I take a similar approach here? I think that would be better than calling something an imagerySet here and using different terminology in the documents.

image

Comment (scottgerring) [reply README.md]: It shouldn't use products in the code; that's wrong too. Overloading domain terminology breaks everyone's minds. 
Ack not introducing a new term as you rightly point out but think it should go `composite product` then.

Comment (ShaunVanDyk) [reply README.md]: Soudns good

**Pull request merged**